test/e2e/query_test.go

```diff
@@ -235,6 +235,39 @@ func TestSidecarNotReady(t *testing.T) {
 	}))
 }
 
+func TestSidecarNotReadyNoPrometheus(t *testing.T) {
+	t.Parallel()
+
+	e, err := e2e.NewDockerEnvironment("sidecar-noProm")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, e))
+
+	// Start sidecar without ever starting Prometheus.
+	_, sidecar := e2ethanos.NewPrometheusWithSidecar(e, "alone", e2ethanos.DefaultPromConfig("prom-alone", 0, "", "", e2ethanos.LocalPrometheusTarget), "", e2ethanos.DefaultPrometheusImage(), "")
+	testutil.Ok(t, sidecar.Start())
```

Comment:
maybe use `e2e.StartAndWaitReady()` here, relying on sidecar's health endpoint as readiness check to make sure we have a running sidecar before testing?